External/FEXCore/Source/Interface/Core/LookupCache.h

```diff
@@ -28,24 +29,51 @@ class LookupCache {
   uintptr_t End() { return 0; }
 
   uintptr_t FindBlock(uint64_t Address) {
-    auto HostCode = FindCodePointerForAddress(Address);
-    if (HostCode) {
-      return HostCode;
-    } else {
-      auto HostCode = BlockList.find(Address);
-
-      if (HostCode != BlockList.end()) {
-        CacheBlockMapping(Address, HostCode->second);
-        return HostCode->second;
-      } else {
-        return 0;
+    // Try L1, no lock needed
+    auto &L1Entry = reinterpret_cast<LookupCacheEntry*>(L1Pointer)[Address & L1_ENTRIES_MASK];
+    if (L1Entry.GuestCode == Address) {
+      return L1Entry.HostCode;
+    }
+
+    // L2 and L3 need to be locked
+    std::lock_guard<std::recursive_mutex> lk(WriteLock);
+
+    // Try L2
+    auto PageIndex = Address & (VirtualMemSize -1);
+
+    uint64_t PageOffset = Address & (0x0FFF);
+    PageIndex >>= 12;
+    uintptr_t *Pointers = reinterpret_cast<uintptr_t*>(PagePointer);
+    uint64_t LocalPagePointer = Pointers[PageIndex];
+    // Do we a page pointer for this address?
+    if (LocalPagePointer) {
+      // Find there pointer for the address in the blocks
+      auto BlockPointers = reinterpret_cast<LookupCacheEntry*>(LocalPagePointer);
+
+      if (BlockPointers[PageOffset].GuestCode == Address)
+      {
+        L1Entry.GuestCode = Address;
+        return L1Entry.HostCode = BlockPointers[PageOffset].HostCode;
```

Comment:
Is assignment intended here? If yes, let's be explicit about it and move it to a dedicated statement.

Reply:
Yep it is intentional

Reply:
Split the assignment, also removed some dead code that slipped through the slicing of the commits